test/Base_Tests/src/Data/Complex_Helpers.enso

```diff
@@ -0,0 +1,25 @@
+private
+
+import Standard.Base.Nothing
+import Standard.Base.Data.Numbers.Float
+import Standard.Base.Data.Ordering.Ordering
+import Standard.Base.Error.Error
+import Standard.Base.Errors.Illegal_Argument.Illegal_Argument
+import project.Data.Complex.Complex
+
+type Complex_Comparator
+    compare x:Complex y:Complex = if x.re==y.re && x.im==y.im then Ordering.Equal else
+        if x.im==0 && y.im==0 then Ordering.compare x.re y.re else
+            Nothing
+    hash x:Complex = if x.im == 0 then Ordering.hash x.re else
+        7*x.re + 11*x.im
+
+## uses the explicit conversion defined in this private module
+Complex.as_complex_and_float self =
+    self : Complex&Float
+
+## explicit "conversion" of `Complex` to `Float` in a private module
+   used in `as_complex_and_float`
+Float.from (that:Complex) =
```

Comment:
Info for library guys (@jdunkerley, @radeusgd, @GregoryTravis, etc.). You can implement values with _mutiple roles_ even right now. Steps:
- define a `type Complex` in one file
- create another `private` _helper module_ with _explicit conversion_
- define extension method on `Complex` that creates the _mixed in value_ - e.g. `as_complex_and_float`
- when you want to "mix" multiple types into `Complex` - call such extension method

Your review is welcomed.